src/Razor/src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -263,8 +263,13 @@ public static CodeWriter WriteStartInstanceMethodInvocation(this CodeWriter writ
                 string.Format(CultureInfo.InvariantCulture, InstanceMethodFormat, instanceName, methodName));
         }
 
-        public static CodeWriter WriteField(this CodeWriter writer, IList<string> modifiers, string typeName, string fieldName)
+        public static CodeWriter WriteField(this CodeWriter writer, IList<string> suppressWarnings, IList<string> modifiers, string typeName, string fieldName)
```

Comment:
This refactor makes it possible to add supressions directly to a field IR node. This is useful because we generally need suppressions on our generated fields/properties/methods.

I just did fields in this PR, but the next time we touch properties or methods we should do similar cleanup, this is really nice to use compared to the existing approach.

Reply:
For instance if we fix the issues with `@inject` + nullable, we could take a similar approach.